pkg/webui/console/views/device-add/index.js

```diff
@@ -41,18 +40,29 @@ import style from './device-add.styl'
     />
   )
 })
-@connect()
+@connect(function (state) {
+  return {
+    device: state.device.device,
+    appId: selectSelectedApplicationId(state),
+  }
+}, dispatch => ({
+  redirectToList: (appId, deviceId) => dispatch(push(`/console/applications/${appId}/devices/${deviceId}`)),
+}),
+(stateProps, dispatchProps, ownProps) => ({
+  ...stateProps, ...dispatchProps, ...ownProps,
+  redirectToList: deviceId => dispatchProps.redirectToList(stateProps.appId, deviceId),
```

Comment:
nitpicking; not getting real benefits from this, since `appId`  is passed as a prop and is used within the component.